### DISCUSSION TITLE
Type (Semantics) Checker

### DISCUSSION THREAD

**Discussion by nordlow** [vlang/v#11850] [Ideas]
Does `v` have a type/semantic checker that skips optimization and code-generation phases and only shows errors?

I've tried `v help` with master version but it only prints

```
`v help`: unknown help topic provided. Use `v help` for usage information.
```

which gives no clue.

**Comment by dumblob** [comment]
What's the use case? Compilation shall be so fast, that this shouldn't matter in practice.

But of course, building some IDE etc. might need something closer to what you describe - in which case I'd refer you to [V language server](https://github.com/vlang/vls ).

**Comment by nordlow** [comment]
All other fast languages, including D, I know becomes at least 2-4 faster when skipping the codegen phase.

**Comment by dumblob** [comment]
Sure it'll be faster - but do you need it faster than sub-second reaction? Why (what's the use case which can't be covered with V language server)?

**Comment by nordlow** [comment]
If V will be as good as it promises then projects will soon be large enough to motivate this optimization.

**Comment by dumblob** [comment]
>If V will be as good as it promises then projects will soon be large enough to motivate this optimization.

What would be the use case which can't be covered by V language server?

**Comment by nordlow** [comment]
> > If V will be as good as it promises then projects will soon be large enough to motivate this optimization.
> 
> What would be the use case which can't be covered by V language server?

A language server is just as fine, likely faster if it can make use of caching aswell. Preferrably at function level.
